orchestrator/gorc/src/commands/sign_delegate_keys.rs

```diff
@@ -1,43 +1,53 @@
 use crate::application::APP;
 use abscissa_core::{Application, Command, Options, Runnable};
+use cosmos_gravity::query as QueryValSet;
 use gravity_proto::gravity as proto;
+use gravity_proto::gravity::query_client::QueryClient as GravityQueryClient;
+use tonic::transport::Channel;
 
 #[derive(Command, Debug, Default, Options)]
 pub struct SignDelegateKeysCmd {
     // TODO(levi) make event-nonce optional: can be queried from a running chain:
     // #[options(free, help = "sign-delegate-key [ethereum-key-name] [validator-address] (event-nonce)")]
     #[options(
         free,
-        help = "sign-delegate-key [ethereum-key-name] [validator-address] [nonce]"
+        help = "sign-delegate-key [ethereum-key-name] [validator-address] (nonce)"
     )]
     pub args: Vec<String>,
 }
 
 impl Runnable for SignDelegateKeysCmd {
     fn run(&self) {
         let config = APP.config();
+        abscissa_tokio::run(&APP, async {
+            let name = self.args.get(0).expect("ethereum-key-name is required");
+            let key = config.load_clarity_key(name.clone());
 
-        let name = self.args.get(0).expect("ethereum-key-name is required");
-        let key = config.load_clarity_key(name.clone());
+            let val = self.args.get(1).expect("validator-address is required");
+            // TODO(levi) ensure this is a valoper address for the next release
 
-        let val = self.args.get(1).expect("validator-address is required");
-        // TODO(levi) ensure this is a valoper address for the next release
+            let nonce = self.args.get(2).expect("nonce is required");
+            let nonce = nonce.parse().expect("could not parse nonce");
```

Comment:
Somewhere in here, you’ll want to use a match statement to see if the nonce was passed in as an argument. Our recover commands are a reasonable example of something similar; where they initialise the mnemonic variable.

Reply:
One branch of the match should match on the user’s command line arg, and the other should fall back to query